src/variables/components/TypeAheadVariableDropdown.tsx

```diff
@@ -51,107 +56,184 @@ class TypeAheadVariableDropdown extends PureComponent<Props, MyState> {
       menuOpen: null,
       loaded: false,
     }
+
+    // if it's a csv var, loading is instantaneous, so if it is done just set it:
+    if (props.status === RemoteDataState.Done) {
+      defaultState.typedValue = props.selectedValue
+      defaultState.actualVal = props.selectedValue
+      defaultState.loaded = true
+    }
+
+    this.state = defaultState
   }
 
   // set the 'shownValues' after loading, and
   // resets the menuOpen variable
   componentDidUpdate(prevProps, prevState) {
     const prevVals = prevProps.values
-    const {values, selectedValue} = this.props
+    const {values, selectedValue, status} = this.props
+    const {status: prevStatus} = prevProps
```

Comment:
just to combine everything, can you also set `prevVals` in this object (if it's possible). it should be something like:

```ts
const {status: prevStatus, values: prevVals} = prevProps
```